src/parser/HLSTree.cpp

```diff
@@ -619,6 +619,7 @@ HLSTree::PREPARE_RESULT HLSTree::prepareRepresentation(Period* period,
           FreeSegments(period, rep);
           rep->segments_.swap(newSegments);
           rep->startNumber_ = newStartNumber;
+          rep->duration_ = period->duration_;
```

Comment:
Previously this was only being set on the last period (an oversight of HLS multi period implementation) and causing issues with the seeking logic here: https://github.com/xbmc/inputstream.adaptive/blob/74205c0cf1625f016a21d1b924cf26af66e2a8ff/src/common/AdaptiveStream.cpp#L1041

Reply:
This also exposes an issue that I believe is in VP - seeking in to the very last moments of a period (not the last one) causes a big sync issue. In a nutshell: with a 2 period video, 10 minutes long and period 2 starts at 6 minutes, if I seek to 5:59 the OSD timer goes back to zero with a black screen, and I will be waiting for 6 minutes for it to 'catch up' and playback of period 2 to begin. Will log an issue for this once I find an easy to reproduce internet stream